src/js/manager.js

```diff
@@ -22,18 +23,30 @@ class SmolLinksManager {
 		return tab;
 	}
 
+	getPage() {
+		let page = 1;
+		// We use 'pg' instead of 'page' because WordPress reserves that query var
+		let pageQuery = location.search.match(/pg=(\d+)/);
+		if (pageQuery) {
+			page = pageQuery[1];
+		}
+		return page;
+	}
+
 	async showResults(result) {
 		try {
 			let response = await result.json();
 			let html = 'Oops, something unexpected happened';
 
 			if (!response.ok || !response.shlink) {
-				html =
-					'Error: ' +
-					(response.error ||
-						'something went wrong loading shlinks. Try again?');
+				html = `<div class="smol-links-error">
+					<strong>Error: </strong>
+					${(response.error ||
+						'something went wrong loading shlinks. Try again?')}
+				</div>`;
 			} else {
 				html = this.getListHTML(response.shlink.shortUrls.data);
+				html += this.getPaginationHTML(response.shlink.shortUrls.pagination);
```

Comment:
```suggestion
				html = this.getPaginationHTML(response.shlink.shortUrls.pagination) + html;
```

to put pagination at the top of the list where people are more likely to see it